src/Firebase/Factory.php

```diff
@@ -249,6 +252,35 @@ public function withClock(object $clock): self
         return $factory;
     }
 
+    public function createAppCheck(): Contract\AppCheck
+    {
+        $projectId = $this->getProjectId();
+        $serviceAccount = $this->getServiceAccount();
+
+        if ($serviceAccount === null) {
+            throw new RuntimeException('Unable to use AppCheck without credentials');
+        }
+
+        $http = $this->createApiClient([
+            'base_uri' => 'https://firebaseappcheck.googleapis.com/v1/projects/'.$projectId,
+        ]);
+
+        $keySet = new CachedKeySet(
+            'https://firebase.google.com/docs/reference/appcheck/rest/v1/jwks#resource:-publicjwkset',
+            new Client(),
+            new HttpFactory(),
+            CacheManager::getInstance('files'),
```

Comment:
This is what I meant when I said that I'd prefer to use an injected instance of a cache instead of declaring it here ourselves - a file cache is undoubtedly good enough for most installations with a filesystem, but there are environments and/or preferences that would require another implementation